src/exception/types.ts

```diff
@@ -65,7 +102,40 @@ export const isProgramExceptionRecord = (input: any): input is ProgramExceptionR
   );
 };
 
-export const isArrayOf = <T>(input: any[], validator: (_: any) => _ is T): input is T[] => {
+export const isSpecimenExceptionRecord = (input: any): input is SpecimenExceptionRecord => {
+  return (
+    // submitter_specimen_id must exist and be a string
+    'submitter_specimen_id' in input && typeof input.submitter_specimen_id === 'string'
+  );
+};
+
+export const isFollowupExceptionRecord = (input: any): input is FollowUpExceptionRecord => {
+  return (
+    // submitter_followup_id must exist and be a string
+    'submitter_followup_id' in input && typeof input.submitter_followup_id === 'string'
+  );
+};
+
+// type guards
+export const isEntityExceptionRecord = (input: any): input is EntityExceptionRecord => {
+  return (
+    isExceptionRecordCheck(input) &&
+    (isSpecimenExceptionRecord(input) || isFollowupExceptionRecord(input))
+  );
+};
+
+const isExceptionRecord = (input: any): input is ExceptionRecord => isExceptionRecordCheck(input);
+
+export const isProgramExceptionRecord = isExceptionRecord;
```

Comment:
for backwards compatibility, I presume?
```suggestion
// TODO: rename this variable wherever else it's being used
export const isProgramExceptionRecord = isExceptionRecord;
```

Reply:
to make things a little more defined.
`isExceptionRecord` is a base case for all exception records
it just so happens that `isProgramExceptionRecord` hasn't got any additional logic right now so they're one and the same.

in practice reading `isProgramExceptionRecord(...)` to check if something is a program exception record, is more clear than , `isExceptionRecord(...)`

Reply:
🤔 clever. I see your point. Likely worth adding a comment about, so no hyperactive and overachieving future Justin will misunderstand the purpose and refactor it away!